qiita_pet/handlers/admin_processing_job.py

```diff
@@ -109,21 +110,43 @@ def post(self):
         # Get user-inputted qiita id and sample names
         qid = self.get_argument("qid")
         snames = self.get_argument("snames").split()
-        error, matching, missing, extra, blank = [None]*5
+        error, matching, missing, extra, blank, duplicates = [None]*6
 
         # Stripping leading qiita id from sample names
         # Example: 1.SKB1.640202 -> SKB1.640202
         try:
-            qsnames = list(Study(qid).sample_template)
+            st = Study(qid).sample_template
+            qsnames = list(st)
         except TypeError:
             error = f'Study {qid} seems to have no sample template'
         except QiitaDBUnknownIDError:
             error = f'Study {qid} does not exist'
 
         if error is None:
+            # if tube_id is present then this should take precedence in qsnames
+            tube_ids = dict()
+            if "tube_id" in st.categories:
+                for k, v in st.get_category("tube_id").items():
+                    # ignoring empty values
+                    if v in (None, 'None', ''):
```

Comment:
Should this be normalized on create/update of the sample template?

Reply:
To be honest, not sure but I think this is something we can improve depending on usage. I decided to use those values as just doing [this](https://github.com/qiita-spots/qiita/pull/3295/files#diff-bcfe36e4beb62086a028b012a89b4432f0812e664ff2214d1d9b7559a26c700bR69); basically returned 'None' in the not updated samples AKA I'm trying to be extra careful.